test/jats-reader.native

```diff
@@ -366,6 +375,428 @@ Pandoc
+      ]
+  , BlockQuote
+      [ Para
+          [ Str "My" , Space , Str "block" , Space , Str "quote" ]
+      ]
+  , Para
+      [ Str "End"
+      , Space
+      , Str "of"
+      , Space
+      , Str "a"
+      , Space
+      , Str "paragraph"
+      ]
+  , Header 1 ( "formulae" , [] , [] ) [ Str "Formulae" ]
+  , Header
+      2
+      ( "inline-formulae" , [] , [] )
+      [ Str "Inline"
+      , Space
+      , Str "formulae"
+      , Space
+      , Str "with"
+      , Space
+      , Emph [ Str "inline-formula" ]
+      ]
+  , Para
+      [ Strong
+          [ Str "Inside" , Space , Str "a" , Space , Str "paragraph" ]
+      , Str ","
+      , Space
+      , Str "should"
+      , Space
+      , Str "be"
+      , Space
+      , Str "natively"
+      , Space
+      , Str "coded"
+      , Space
+      , Str "inside"
+      , Space
+      , Str "the"
+      , Space
+      , Str "paragraph,"
+      , Space
+      , Str "and"
+      , Space
+      , Str "show"
+      , Space
+      , Str "inline:"
+      ]
+  , Para
+      [ Str "Einstein"
+      , Space
+      , Str "showed"
+      , Space
+      , Str "that"
+      , Space
+      , Math InlineMath "e=mc^2"
+      , Str "."
+      , SoftBreak
+      , Str "This"
+      , Space
+      , Str "formula"
+      , Space
+      , Str "is"
+      , Space
+      , Str "important"
+      , Space
+      , Str "because\8230"
+      ]
+  , Header
+      2
+      ( "disp-formulae" , [] , [] )
+      [ Str "Block"
+      , Space
+      , Str "formulae"
+      , Space
+      , Str "with"
+      , Space
+      , Emph [ Str "disp-formula" ]
+      ]
+  , Para
+      [ Strong
+          [ Str "Inside" , Space , Str "a" , Space , Str "paragraph" ]
+      , Str ","
+      , Space
+      , Str "should"
+      , Space
+      , Str "be"
+      , Space
+      , Str "natively"
+      , Space
+      , Str "coded"
+      , Space
+      , Str "inside"
+      , Space
+      , Str "the"
+      , Space
+      , Str "pragraph,"
+      , Space
+      , Str "but"
+      , Space
+      , Str "show"
+      , Space
+      , Str "as"
+      , Space
+      , Str "block:"
+      ]
+  , Para
+      [ Str "Einstein"
+      , Space
+      , Str "showed"
+      , Space
+      , Str "that"
+      , Space
+      , Math DisplayMath "e=mc^2"
+      , Str "."
+      , SoftBreak
+      , Str "This"
+      , Space
+      , Str "formula"
+      , Space
+      , Str "is"
+      , Space
+      , Str "important"
+      , Space
+      , Str "because\8230"
+      ]
+  , Para
+      [ Strong
+          [ Str "Outside" , Space , Str "a" , Space , Str "pararaph" ]
+      , Str ","
+      , Space
+      , Str "should"
+      , Space
+      , Str "show"
+      , Space
+      , Str "as"
+      , Space
+      , Str "block:"
+      ]
+  , Para
+      [ Str "Einstein"
+      , Space
+      , Str "showed"
+      , Space
+      , Str "that"
+      ]
+  , Div
+      ( "" , [ "disp-formula" ] , [] )
+      [ LineBlock [ [ Math DisplayMath "e=mc^2" ] ] ]
```

Comment:
Why the LineBlock?
I wonder whether the Div with class "disp-formula" is really needed.  Why not just a LineBlock or even just a Para with the Math DisplayMath inside it?

Reply:
This is because in this case the `disp-formula` is not the inmediata parent of the math elements, but the grandparent.  So the `disp-formula` builds the Div, then the `alternatives` parent builds the LineBlock. 

Also, I suppose it's a balance between a clean output and a clean code. The only way of getting rid of the Div is to start adding what can become convoluted conditions to the parse cases for `disp-formula` and `alternatives`, while right now they are quite simple, the recursion does the rest. 

But I see the point. I could definitely give it more thought and see in the following days if I find a way to write the added conditions nicely.

Reply:
Simple is good.  But why 

> the alternatives parent builds the LineBlock.

What does LineBlock have to do with alternatives?

Reply:
LineBlock has to do with whoever is the direct parent of math elements. `alternatives` does not always build a LineBlock, it does only when it is the parent of `mml:math` or `tex-math`.

Conversely, `disp-math` doesn't always build a Div. It does when it is not the direct parent of math elements; but if it is, it too builds a LineBlock.

Reply:
Okay, but then my question is: what does LineBlock have to do with mml:math or tex-math?

Reply:
There is no specific or intrinsic relationship between LineBlock and math elements. We could have chosen another type of block, like Div. 

LineBlock fitted nicely with the code for  `blockFormula`, a new function derived from `inlineFormula` (it has the same input and output type as `mconcat`, the function it replaces in the block version of the formula builder).

Reply:
I think Para is the normal default choice for display math that is in a block by itself.  Unless there's a good specific reason to use LineBlock, we should use Para.

Reply:
OK, fair enough. Just replaced LineBlock by Para.